openwisp_notifications/tasks.py

```diff
@@ -202,3 +211,70 @@ def delete_ignore_object_notification(instance_id):
     Deletes IgnoreObjectNotification object post it's expiration.
     """
     IgnoreObjectNotification.objects.filter(id=instance_id).delete()
+
+
+@shared_task(base=OpenwispCeleryTask)
+def batch_email_notification(email_id):
+    """
+    Sends a summary of notifications to the specified email address.
+    """
+    ids = cache.get(f'{email_id}_batch_pks', [])
+
+    if not ids:
+        return
+
+    unsent_notifications = Notification.objects.filter(id__in=ids)
+    notifications_count = unsent_notifications.count()
+    current_site = Site.objects.get_current()
+
+    if notifications_count == 1:
+        instance = unsent_notifications.first()
+        send_notification_email(instance)
+    else:
+        alerts = []
+        for notification in unsent_notifications:
+            url = notification.data.get('url', '') if notification.data else None
+            if url:
+                target_url = url
+            elif notification.target:
+                target_url = notification.redirect_view_url
+            else:
+                target_url = None
+
+            description = notification.message if notifications_count <= 5 else None
+            alerts.append(
+                {
+                    'level': notification.level,
+                    'message': notification.message,
+                    'description': description,
+                    'timestamp': notification.timestamp,
+                    'url': target_url,
+                }
+            )
+
+        context = {
+            'alerts': alerts,
+            'notifications_count': notifications_count,
+            'site_name': current_site.name,
+            'site_domain': current_site.domain,
+        }
+        html_content = render_to_string('emails/batch_email.html', context)
+        plain_text_content = strip_tags(html_content)
```

Comment:
The plain text email looks like this 

```
Content-Type: multipart/alternative;
 boundary="===============8141022284195897303=="
MIME-Version: 1.0
Subject: Summary of 3 Notifications
From: webmaster@localhost
To: admin@admin.com
Date: Tue, 18 Jun 2024 07:30:02 -0000
Message-ID: <171869580268.41107.14558799017281335812@FRIDAY>

--===============8141022284195897303==
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit



        .container-alerts {
            margin: 0 auto;
            background-color: #fff;
            padding: 20px;
            border: 1px solid #ddd;
            box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
        }
        .header {
            text-align: center;
            padding: 10px 0;
            border-bottom: 1px solid #ddd;
            margin-bottom: 20px;
        }
        .header h1 {
            margin: 0;
            font-size: 24px;
        }
        .header p {
            margin: 5px 0 0 0;
            font-size: 14px;
            color: #666;
        }
        .alert {
            border: 1px solid #e0e0e0;
            border-radius: 5px;
            margin-bottom: 10px;
            padding: 10px;
        }
        .alert.error {
            background-color: #ffefef;
        }
        .alert.info {
            background-color: #f0f0f0;
        }
        .alert.success {
            background-color: #e6f9e8;
        }
        .alert h2 {
            margin: 0 0 5px 0;
            font-size: 16px;
        }
        .alert h2 .title {
            display: inline-block;
            max-width: 80%;
            white-space: nowrap;
            overflow: hidden;
            text-overflow: ellipsis;
            vertical-align: middle;
        }
        .alert.error h2 {
            color: #d9534f;
        }
        .alert.info h2 {
            color: #333333;
        }
        .alert.success h2 {
            color: #1c8828;
        }
        .alert p {
            margin: 0;
            font-size: 14px;
            color: #666;
        }
        .alert .title p {
            display: inline;
            overflow: hidden;
            text-overflow: ellipsis;
        }
        .badge {
            display: inline-block;
            padding: 2px 8px;
            border-radius: 3px;
            font-size: 12px;
            font-weight: bold;
            text-transform: uppercase;
            margin-right: 5px;
            color: white;
        }
        .badge.error {
            background-color: #d9534f;
        }
        .badge.info {
            background-color: #333333;
        }
        .badge.success {
            background-color: #1c8828;
        }
        .alert a {
            text-decoration: none;
        }
        .alert.error a {
            color: #d9534f;
        }
        .alert.info a {
            color: #333333;
        }
        .alert.success a {
            color: #1c8828;
        }
        .alert a:hover {
            text-decoration: underline;
        }






            Summary of 3 Notifications
            From example.com




                INFO


                        Default notification with default verb and level info by default



            June 18, 2024, 9:29 a.m.

                Default notification with default verb and level info by default





                INFO


                        Default notification with default verb and level info by default



            June 18, 2024, 9:29 a.m.

                Default notification with default verb and level info by default





                INFO


                        Default notification with default verb and level info by default



            June 18, 2024, 9:29 a.m.

                Default notification with default verb and level info by default

```

It shall not contain the CSS block. The `strip_tags` is not removing them from the rendered string. We may need to add a dedicated template for the plain text email. 

Let's discuss this in today's weekly meeting before proceeding.